### PR TITLE
Feature/timeline read receipts

### PR DESCRIPTION
**Pull request by ganfra**
This PR introduces Read Receipts on timeline. We can hide them from settings/preferences screen.

Screenshot_20190814-105335
Screenshot_20190814-105349

### Pull Request Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 16
- [x] UI change has been tested on both light and dark themes
- [x] Pull request is based on the develop branch
- [x] Pull request updates [CHANGES.md](https://github.com/vector-im/riotX-android/blob/develop/CHANGES.md)
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/timeline/TimelineHiddenReadReceipts.kt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import im.vector.matrix.android.internal.database.model.TimelineEventEntity
+import im.vector.matrix.android.internal.database.model.TimelineEventEntityFields
+import im.vector.matrix.android.internal.database.query.FilterContent
+import im.vector.matrix.android.internal.database.query.whereInRoom
+import io.realm.OrderedRealmCollectionChangeListener
+import io.realm.Realm
+import io.realm.RealmQuery
+import io.realm.RealmResults
+
+internal class TimelineHiddenReadReceipts constructor(private val readReceiptsSummaryMapper: ReadReceiptsSummaryMapper,
```
Comment (BillCarsonFr) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/timeline/TimelineHiddenReadReceipts.kt]: some comments to explain the 'hidden' read receipts things?

File: CHANGES.md [hunk 2]
```diff
@@ -2,7 +2,7 @@ Changes in RiotX 0.4.0 (2019-XX-XX)
 ===================================================
 
 Features:
- -
+ - Display read receipts in timeline
```
Comment (bmarty) [inline CHANGES.md]:
```suggestion
 - Display read receipts in timeline (#81)
```

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/timeline/TimelineEvent.kt [hunk 3]
```diff
@@ -94,7 +96,7 @@ fun TimelineEvent.hasBeenEdited() = annotations?.editSummary != null
  * Get last MessageContent, after a possible edition
  */
 fun TimelineEvent.getLastMessageContent(): MessageContent? = annotations?.editSummary?.aggregatedContent?.toModel()
-        ?: root.getClearContent().toModel()
+                                                             ?: root.getClearContent().toModel()
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/api/session/room/timeline/TimelineEvent.kt]: I'm fed up with all this formatting diffs. Can we share the same formatting rules please?

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/query/FilterContent.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.matrix.android.internal.database.query
+
+internal object FilterContent {
+
+    internal const val EDIT_TYPE = "{*\"m.relates_to\"*\"rel_type\":*\"m.replace\"*}"
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/database/query/FilterContent.kt]: Use `"""` for code clarity

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/ReadReceiptHandler.kt [hunk 5]
```diff
@@ -29,34 +33,70 @@ import javax.inject.Inject
...
+        } else {
+            incrementalSyncStrategy(realm, roomId, content)
+        }
+    }
+
+
+    private fun initialSyncStrategy(realm: Realm, roomId: String, content: ReadReceiptContent) {
+        val readReceiptSummaries = ArrayList<ReadReceiptsSummaryEntity>()
+        for ((eventId, receiptDict) in content) {
+            val userIdsDict = receiptDict[READ_KEY] ?: continue
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/ReadReceiptHandler.kt]: `filterKeys {}` was not working?

File: vector/src/main/res/layout/item_display_read_receipt.xml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    <TextView
+        android:id="@+id/readReceiptName"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="4dp"
+        android:layout_marginLeft="4dp"
+        android:layout_weight="1"
+        android:ellipsize="end"
+        android:lines="1"
+        android:textColor="?android:textColorPrimary"
```
Comment (bmarty) [inline vector/src/main/res/layout/item_display_read_receipt.xml]: Use `?riotx_text_primary`

File: vector/src/main/res/layout/item_display_read_receipt.xml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        android:textSize="16sp"
+        tools:text="@sample/matrix.json/data/displayName" />
+
+    <TextView
+        android:id="@+id/readReceiptDate"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:lines="1"
+        android:textColor="?android:textColorSecondary"
```
Comment (bmarty) [inline vector/src/main/res/layout/item_display_read_receipt.xml]: Use `?riotx_text_secondary`

File: vector/src/main/res/layout/item_display_read_receipt.xml [hunk 8]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="44dp"
```
Comment (bmarty) [inline vector/src/main/res/layout/item_display_read_receipt.xml]: I do not like hard-coded height on a cell containing text. If user choose to use big font, the render will not be correct.
Instead use `wrap_content` with padding on text for instance

**Pull request merged**